### PR TITLE
differentiate between browser and node builds

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "@stellar/typescript-wallet-sdk",
-  "version": "1.1.0",
+  "version": "1.1.1",
   "engines": {
     "node": ">=18"
   },
+  "browser": "./lib/bundle_browser.js",
   "main": "./lib/bundle.js",
```
Comment (piyalbasu) [inline package.json:8]: bundlers (like `webpack` and `rollup`) use the file specified in the `browser` field to build for web apps while all other apps default to the `main` field. 

So, we'll point to a specific browser build for web apps while defaulting to a node version

https://docs.npmjs.com/cli/v9/configuring-npm/package-json#browser

File: webpack.config.js [hunk 2]
```diff
@@ -25,10 +25,11 @@ module.exports = {
     },
   },
   output: {
-    library: "MyLibrary",
+    library: "WalletSDK",
```
Comment (piyalbasu) [inline webpack.config.js:28]: changing this to be more descriptive

File: webpack.config.js [hunk 3]
```diff
@@ -25,10 +25,11 @@ module.exports = {
...
     libraryTarget: "umd",
     globalObject: "this",
-    filename: "bundle.js",
+    filename: `bundle${!env.NODE ? "_browser" : ""}.js`,
     path: path.resolve(__dirname, "lib"),
   },
-};
+  target: env.NODE ? "node" : "web",
```
Comment (piyalbasu) [inline webpack.config.js:34]: here we generate 2 different builds based on a webpack flag

File: package.json [hunk 4]
```diff
@@ -47,6 +49,8 @@
   "scripts": {
     "prepare": "yarn build",
     "test": "jest --watchAll",
-    "build": "webpack --config webpack.config.js"
+    "build:web": "webpack --config webpack.config.js",
+    "build:node": "webpack --env NODE=true --config webpack.config.js",
+    "build": "run-p build:web build:node"
```
Comment (piyalbasu) [inline package.json:54]: our build job will build for both web and node in parallel

**Review by Ifropc — APPROVED** [review]

**Comment by acharb** [comment]
lgtm! thanks for the quick fixing!

**Pull request merged**